src/components/karaoke.tsx

```diff
@@ -38,14 +37,15 @@ export function Karaoke({ code }: { code: string }) {
       id: string;
     }>
   >([]);
-  // const [currentVideo, setCurrentVideo] = useState<null | Document>(null);
   const [currentVideo, setCurrentVideo] = useState<Video | null>(null);
   const [editingIndex, setEditingIndex] = useState(null);
   const [newName, setNewName] = useState("");
   const [newSong, setNewSong] = useState("");
   const [newVideoUrl, setNewVideoUrl] = useState("");
   const [newNameError, setNewNameError] = useState("");
   const [newVideoUrlError, setNewVideoUrlError] = useState("");
+  const [loading, setLoading] = useState(true);
+  const [error, setError] = useState<string | null>(null);
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The state variables `newName`, `newSong`, `newVideoUrl`, `newNameError`, and `newVideoUrlError` are initialized with empty strings. It would be better to initialize them as `null` or `undefined` to clearly indicate that they have not been set yet. This can help avoid potential bugs related to type coercion in JavaScript.
```diff
-   const [newName, setNewName] = useState("");
-   const [newSong, setNewSong] = useState("");
-   const [newVideoUrl, setNewVideoUrl] = useState("");
-   const [newNameError, setNewNameError] = useState("");
-   const [newVideoUrlError, setNewVideoUrlError] = useState("");
+   const [newName, setNewName] = useState(null);
+   const [newSong, setNewSong] = useState(null);
+   const [newVideoUrl, setNewVideoUrl] = useState(null);
+   const [newNameError, setNewNameError] = useState(null);
+   const [newVideoUrlError, setNewVideoUrlError] = useState(null);
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->